vendor/psmodules/Cmder.ps1

```diff
@@ -1,178 +1,204 @@
-function readGitVersion($gitPath) {
-    $gitExecutable = "${gitPath}\git.exe"
+function Get-GitVersion {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$GitPath
+    )
 
-    if (-not (Test-Path "$gitExecutable")) {
-        return $null
-    }
-
-    $gitVersion = (cmd /c "${gitExecutable}" --version)
+    $gitExecutable = Join-Path $GitPath "git.exe"
 
-    if ($gitVersion -match 'git version') {
-        ($trash1, $trash2, $gitVersion) = $gitVersion.split(' ', 3)
-    } else {
-        pause
+    if (-not (Test-Path $gitExecutable)) {
         return $null
     }
 
-    return $gitVersion.toString()
-}
-
-function isGitShim($gitPath) {
-    # Check if there is a shim file - if yes, read the actual executable path
-    # See: github.com/ScoopInstaller/Shim
+    $gitVersion = & $gitExecutable --version 2>$null
 
-    if (Test-Path "${gitPath}\git.shim") {
-        $shim = (get-content "${gitPath}\git.shim")
-        ($trash, $gitPath) = $shim.replace(' ', '').split('=')
-
-        $gitPath = $gitPath.replace('\git.exe', '')
+    if ($gitVersion -match 'git version\s+(\S+)') {
+        return $Matches[1]
     }
 
-    return $gitPath.toString()
+    Write-Debug "Git executable path: $gitExecutable"
+    Write-Error "'git --version' returned an improper version string!"
+    Write-Error "Unable to determine Git version from output: $gitVersion"
+
+    return $null
 }
 
-function compareVersions($userVersion, $vendorVersion) {
-    if ($null -ne $userVersion) {
-        ($userMajor, $userMinor, $userPatch, $userBuild) = $userVersion.split('.', 4)
-    } else {
-        return -1
+function Get-GitShimPath {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$GitPath
+    )
+
+    $shimFile = Join-Path $GitPath "git.shim"
+    if (Test-Path $shimFile) {
+        $shimContent = Get-Content $shimFile -Raw
+        if ($shimContent -match 'path\s*=\s*(.+)') {
+            $GitPath = $Matches[1].Trim().Replace('\git.exe', '')
+        }
     }
 
-    if ($null -ne $vendorVersion) {
-        ($vendorMajor, $vendorMinor, $vendorPatch, $vendorBuild) = $vendorVersion.split('.', 4)
-    } else {
-        return 1
-    }
+    return $GitPath
+}
 
-    if (($userMajor -eq $vendorMajor) -and ($userMinor -eq $vendorMinor) -and ($userPatch -eq $vendorPatch) -and ($userBuild -eq $vendorBuild)) {
-        return 1
+function Compare-Version {
+    param(
+        [Parameter(Mandatory = $false)]
+        [AllowNull()]
+        [string]$UserVersion,
+        [Parameter(Mandatory = $false)]
+        [AllowNull()]
+        [string]$VendorVersion
+    )
+
+    if ($null -eq $UserVersion) { return -1 }
+    if ($null -eq $VendorVersion) { return 1 }
+
+    try {
+        $userVer = [version]$UserVersion
+        $vendorVer = [version]$VendorVersion
```

Comment:
Good idea, using the native `[version]` class, instead of manually parsing and comparing them! 👍🏻